### PR TITLE
81 add esma register

### PR DESCRIPTION
**Pull request by SimonThordal**
https://registers.esma.europa.eu/publication/searchRegister?core=esma_registers_saris_new

This includes EU sanctioned entities. Gold mine!

### CODE REVIEW

File: datasets/eu/esma_saris/crawler.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            if isin is None:
+                context.log.warn("No ISIN", row=row)
+                return
+            entity.id = f"isin-{isin}"
+            entity.add("isin", isin)
+            entity.add("name", row.pop("instrumentFullName", isin))
+
+            sanction = helpers.make_sanction(context, entity)
+            sanction.id = context.make_id(row.pop("id"))
+            sanction.add("program", "ESMA")
```
Comment (pudo) [inline datasets/eu/esma_saris/crawler.py:38]: I don't think we need this: the sanction['authority'] is already going to contain the `publisher.name`.

File: datasets/eu/esma_saris/crawler.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            if row.get("effectiveTo") is not None:
+                sanction.add("endDate", row.pop("effectiveTo"))
+            else:
+                entity.add("topic", "sanction")
+            context.emit(sanction, target=False)
+
+            if row.get("issuer") is not None:
+                issuer = context.make("LegalEntity")
+                issuer.id = context.make_id(row.pop("issuer"))
+                issuer.add("leiCode", issuer)
```
Comment (pudo) [inline datasets/eu/esma_saris/crawler.py:54]: If the issuer has a LEI, there's also a fake ID we could use: `lei-{lei}`. It's already being emitted by 3-4 other sources.

File: datasets/eu/esma_saris/crawler.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+            sanction = helpers.make_sanction(context, entity)
+            sanction.id = context.make_id(row.pop("id"))
+            sanction.add("program", "ESMA")
+            sanction.add("provisions", row.pop("actionType"))
+            sanction.add("reason", row.pop("reasonsForTheAction"))
+            sanction.add("description", row.pop("comments"))
+            sanction.add("startDate", row.pop("effectiveFrom"))
+            sanction.add("country", row.pop("memberStateOfNotifyingCA"))
+            sanction.add("authority", row.pop("notifyingCA"))
```
Comment (pudo) [inline datasets/eu/esma_saris/crawler.py:44]: Ah, so you may want to make this a `.set(`, replacing the previous value.

**Review by pudo — COMMENTED** [review]
Looks really good!

File: datasets/eu/esma_saris/crawler.py (hunk 1 above)
Comment (SimonThordal) [reply datasets/eu/esma_saris/crawler.py:38]: Yeah, this is a bit of a weird one. As far as I can understand ([basis here](https://www.esma.europa.eu/publications-and-data/interactive-single-rulebook/mifid-ii/article-52-suspension-and-removal)) ESMA is sort of an "aggregator" of suspensions and removals.

Is the program the name of directive itself maybe? Or do we leave it blank

File: datasets/eu/esma_saris/crawler.py (hunk 2 above)
Comment (SimonThordal) [reply datasets/eu/esma_saris/crawler.py]: Looks like it only sometimes has a LEI and sometimes just has a 4 letter random string, so sometimes the `lei-` will work, sometimes it won't, so I made it conditional.

File: datasets/eu/esma_saris/crawler.py (hunk 3 above)
Comment (SimonThordal) [reply datasets/eu/esma_saris/crawler.py]: Thanks, fixed!

**Pull request merged**